src/KilnUniV2LPSwap.sol

```diff
@@ -56,42 +56,109 @@ interface UniswapRouterV2Like {
     ) external returns (uint amountA, uint amountB);
 }
 
+interface UniswapPairV2Like {
+    function getReserves() external view returns (uint112, uint112, uint32);
+}
+
+interface PipLike {
+    function peek() external returns (bytes32, bool);
+}
+
 contract KilnUniV2LPSwap is KilnBase {
 
+    event Debug(uint256, uint256);
+
     using UniswapV2Library for *;
 
     address public immutable uniV2Router;
     address public immutable pairToken;
     address public immutable receiver;
 
-    event LogI(uint256);
+    uint256 internal constant WAD = 10 ** 18;
+
+    uint256 public max;  // [WAD] Maximum acceptable buy price in sell.
```

Comment:
Conceptually, I think this is a minimum, not a maximum? It is related by direct proportionality to `amountOutMin`.

Reply:
Maybe needs a more expressive variable. This is the max price the kiln is willing to pay in `buy` token for `sell` token.